src/x86/intel_edison_fab_c.c

```diff
@@ -373,8 +374,17 @@ mraa_intel_edison_aio_init_post(mraa_aio_context dev)
 mraa_result_t
 mraa_intel_edison_pwm_enable_pre(mraa_pwm_context dev, int enable) {
     // PWM 0% duty workaround: update state array
-    // if someone first ran write(0) and then enable(1).
-    if ((pwm_disabled[dev->pin] == 1) && (enable == 1)) { pwm_disabled[dev->pin] = 0; }
+    if (enable == 1) {
+        pwm_disabled[dev->pin] = 0;
+    } else {
+        pwm_disabled[dev->pin] = 1;
+        // Set duty cycle to 0 before disable pwm.
+        // Edison PWM output stuck at high if disabled during ON period
+        mraa_pwm_pulsewidth_us(dev, 0);
+        // Sleep 2 period to allow change take effect
+        usleep(dev->period >> 9);
```

Comment:
I agree the delay is not desirable, do you recommend closing this PR?

From my view your test result mostly agree with this observation. 100% failure for 100%duty case, and 0% failure for 10% duty case. The lack of smooth transition could be caused by deterministic timing between operations.

For the last part, I agree it's unlikely going to make any difference, both in terms of speed as well as the timing (from my test, I stopped seeing any failure once delay reaches ~2 * period), so I went with my first instinct. I'll change this to the verbose way.